src/api/CartApi.js

```diff
@@ -0,0 +1,93 @@
+import axios from "axios";
+
+const getUserCartList = async (setUserCartList) => {
+    try{
+        const response = await axios.get("/api/user/cart",{
+            headers : {
+                authorization : localStorage.getItem("token")
+            }
+        })
+        setUserCartList(response.data.cart)
+    }catch({response}){
+        console.log(response)
+    }
+}
+
+const addToUserCart = async (product,dispatchCart) => {
+    const encodedToken = localStorage.getItem("token");
+    try {
+        if (encodedToken) {
+            await axios.post("/api/user/cart",  {product},
+            { headers: { authorization: encodedToken } },
+           )
+            dispatchCart({
+                operation: "ADD_TO_CART",
+                payLoad: product
+            })
+        }
+        else {
+            dispatchCart({
+                operation: "ADD_TO_CART",
+                payLoad: product
+            })
+        }
```

Comment:
Can you explain this

Reply:
it's just if user is not logged it than the same method will do the carddispatch